ts/features/itwallet/common/utils/ItwStoreUtils.ts

```diff
@@ -0,0 +1,45 @@
+import { GlobalState } from "../../../../store/reducers/types";
+
+interface PollForStoreValueOptions<T> {
+  state: GlobalState;
+  selector: (state: GlobalState) => T;
+  condition: (value: T) => boolean;
+  interval?: number;
+  timeout?: number;
+}
+
+/**
+ * Polls the Redux store until the selected value meets the specified condition.
+ * @param state The Redux store state.
+ * @param selector A function that selects the value from the store.
+ * @param condition A function that checks if the value meets the desired condition.
+ * @param interval The interval in milliseconds between each check.
+ * @param timeout The maximum time in milliseconds to wait for the condition to be met.
+ * @returns A promise that resolves with the store value once the condition is met.
+ */
+export const pollForStoreValue = <T>({
+  state,
+  selector,
+  condition,
+  interval = 1000,
+  timeout = 10000
+}: PollForStoreValueOptions<T>): Promise<T> =>
+  new Promise((resolve, reject) => {
+    const startTime = Date.now();
+
+    const checkCondition = () => {
+      const value = selector(state);
+
+      if (condition(value)) {
+        resolve(value);
+        return;
+      }
+
+      if (Date.now() - startTime > timeout) {
+        reject(new Error("Timeout exceeded while waiting for store value"));
+        return;
+      }
+      setTimeout(checkCondition, interval);
+    };
+    checkCondition();
+  });
```

Comment:
A couple of suggestions:
- we should pass a function to get the state instead of passing a static state object. This allows to retrieve the latest state on each polling attempt.
- using a `setInterval` could be a cleaner way to manage the retries, instead of recursively call `setTimeout`.
What do you think? Here is an example:

```suggestion
interface PollForStoreValueOptions<T> {
  getState: () => GlobalState;
  selector: (state: GlobalState) => T;
  condition: (value: T) => boolean;
  interval?: number;
  timeout?: number;
}

/**
 * Polls the Redux store until the selected value meets the specified condition.
 * @param state The Redux store state.
 * @param selector A function that selects the value from the store.
 * @param condition A function that checks if the value meets the desired condition.
 * @param interval The interval in milliseconds between each check.
 * @param timeout The maximum time in milliseconds to wait for the condition to be met.
 * @returns A promise that resolves with the store value once the condition is met.
 */
export const pollForStoreValue = <T>({
  getState,
  selector,
  condition,
  interval = 1000,
  timeout = 10000
}: PollForStoreValueOptions<T>): Promise<T> =>
  new Promise((resolve, reject) => {
    const startTime = Date.now();
        
      const intervalId = setInterval(() => {
        const currentTime = Date.now();
        const value = selector(getState());
  
        if (condition(value)) {
          clearInterval(intervalId);
          resolve(value);
        } else if (currentTime - startTime >= timeout) {
          clearInterval(intervalId);
          reject(new Error("Timeout exceeded while waiting for store value"));
        }
      }, interval);
    });
```

Reply:
Thank you @mastro993  for the suggestion! Your solution is more efficient and better than mine. I implemented it and also added tests inherent to this polling function.